ntbs-service/wwwroot/source/Components/FilteredDropdown.ts

```diff
@@ -0,0 +1,122 @@
+import Vue from "vue";
+import { getHeaders } from "../helpers";
+import axios from "axios";
+
+type OptionValue = {
+    value: string,
+    text: string,
+    group: string
+};
+
+const FilteredDropdown = Vue.extend({
+    props: {
+        filterHandlerPath: {
+            type: String
+        },
+        filteringRefs: {
+            type: Array
+        },
+        waitForChildMount: {
+            type: Boolean,
+            default: true
+        }
+    },
+    mounted: function () {
+        if (!this.waitForChildMount) {
+            this.filteringMounted();
+        }
+    },
+    methods: {
+        filteringMounted: function () {
+            this.fetchFilteredLists(this.getFilteringValue());
+        },
+        filteringChanged: function () {
+            this.fetchFilteredLists(this.getFilteringValue());
+        },
+        getFilteringValue: function () {
+            return this.getSelectElementInRef("filterContainer").value;
+        },
+        fetchFilteredLists: function (value: string) {
+            const requestConfig = {
+                url: `${window.location.pathname}/${this.filterHandlerPath}`,
+                headers: getHeaders(),
+                params: {
+                    "value": value
+                }
+            }
+
+            axios.request(requestConfig)
+                .then((response: any) => {
```

Comment:
Is it worth giving the response a type?

Reply:
Think we could only specify it down to `{ data: any }` - so not sure it's worth it.
Currently data either contains `{Hospital: [], CaseManagers:[]}` or `{SampleTypes:[]}`